pkg/security/security_profile/profile/manager.go

```diff
@@ -511,62 +531,73 @@ func (m *SecurityProfileManager) unlinkProfile(profile *SecurityProfile, workloa
 	seclog.Infof("workload %s (selector: %s) successfully unlinked from profile %s", workload.ID, workload.WorkloadSelector.String(), profile.Metadata.Name)
 }
 
-func (m *SecurityProfileManager) LookupEventOnProfiles(event *model.Event) {
-	evtType := event.GetEventType()
-	if evtType == model.SyscallsEventType || // syscall matching for anomaly detection is already done kernel side
-		evtType == model.FileOpenEventType || evtType == model.BindEventType || // disabled for now
-		evtType == model.ForkEventType || evtType == model.ExitEventType { // no interest in fork/exit events
-		return
-	}
-
-	if event.Error != nil {
-		m.eventFilteringAbsent[evtType].Inc()
+func (m *SecurityProfileManager) LookupEventInProfiles(event *model.Event) {
+	// shortcut for dedicated anomaly detection events
```

Comment:
I think we should also have some shortcuts here to check first if the event is a valid one.
You move both checks (event types and errors) on profile:isEventValid(), which is only call later on when trying to insert or checking if the event is present on a profile.
This could avoid unneeded calls to resolvers, workloadselector creation, profiles lookup etc

Reply:
Yep it could be error prone.. Maybe we could call isEventValid() directly from both dump and profile sides?